synapse/handlers/federation.py

```diff
@@ -460,11 +460,44 @@ def _process_received_pdu(self, origin, pdu, state, auth_chain):
     @measure_func("_filter_events_for_server")
     @defer.inlineCallbacks
     def _filter_events_for_server(self, server_name, room_id, events):
+        # First lets check to see if all the events have a history visibility
+        # of "shared" or "world_readable". If thats the case then we don't
```

Comment:
need to spell out why it's safe to not filter for "shared" history visibility (apparently because 'all the places that call it check the server is in the room'?)